app/code/Magento/Catalog/Model/Category/Attribute/Source/Layout.php

```diff
@@ -30,10 +30,9 @@ public function __construct(\Magento\Framework\View\Model\PageLayout\Config\Buil
      */
     public function getAllOptions()
     {
-        if (!$this->_options) {
-            $this->_options = $this->pageLayoutBuilder->getPageLayoutsConfig()->toOptionArray();
-            array_unshift($this->_options, ['value' => '', 'label' => __('No layout updates')]);
-        }
-        return $this->_options;
+        $options = $this->pageLayoutBuilder->getPageLayoutsConfig()->toOptionArray();
+        array_unshift($options, ['value' => '', 'label' => __('No layout updates')]);
+
+        return $options;
```

Comment:
Due to

> Continue storing the value in the property to preserve the old functionality.

Please mark property as deprecated but assign value to it each call.